src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java

```diff
@@ -0,0 +1,45 @@
+package com.qaprosoft.carina.demo;
+
+import com.qaprosoft.carina.core.foundation.IAbstractTest;
+import com.qaprosoft.carina.core.foundation.api.http.HttpResponseStatusType;
+import com.qaprosoft.carina.demo.soap.AddInteger;
+import com.qaprosoft.carina.demo.soap.LookupCity;
+import io.restassured.path.xml.XmlPath;
+import io.restassured.response.Response;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class SoapSampleTest implements IAbstractTest {
+
+    @Test
+    public void testLookupCity() {
+        LookupCity soap = new LookupCity();
+        soap.setProperties("api/soap/soap.properties");
+        soap.setHeaders(String.format("Content-type=%s", "text/xml"));
```

Comment:
I think by calling 'setHeaders' 2 times you will override your first setting. please review how setHeaders() method works.
Also there is annotation ContentType available for AbstractAPIMethod.
Please use it better instead

Reply:
Let's also introduce method addSOAPHeader() and getSOAPHeader()